### PR TITLE
af_lavrresample: better swr reinitialization

### PR DESCRIPTION
**Pull request by richardpl**

### CODE REVIEW

File: audio/filter/af_lavrresample.c [hunk 1]
```diff
@@ -400,10 +400,8 @@ static int control(struct af_instance *af, int cmd, void *arg)
     case AF_CONTROL_RESET:
         if (s->avrctx) {
 #if HAVE_LIBSWRESAMPLE
-            // This POS either can't drop state correctly, or doesn't want to.
-            // It will swallow some minor audio e.g. after a seek.
-            // Deallocate and recreate the resample state for a full reset.
-            configure_lavrr(af, &af->fmt_in, &af->fmt_out, false);
+            swr_close(s->avrctx);
+            swr_init(s->avrctx);
```
Comment (wm4) [inline audio/filter/af_lavrresample.c]: I don't quite trust it. There's way too much going on in swr_init. Are you absolutely sure it doesn't clobber some input options (which, if swr_init() is run with the clobbered value, would result in different parameters)?

Also swr_init can fail, which you should check. (Call close_lavrr on failure.)

**Comment by richardpl** [comment]
There is not much difference between this and full reinitialization, swr_close frees all its internal memory.
I don't see any clobbered values.

File: audio/filter/af_lavrresample.c [hunk 2]
```diff
@@ -400,10 +400,13 @@ static int control(struct af_instance *af, int cmd, void *arg)
...
-            // This POS either can't drop state correctly, or doesn't want to.
-            // It will swallow some minor audio e.g. after a seek.
-            // Deallocate and recreate the resample state for a full reset.
-            configure_lavrr(af, &af->fmt_in, &af->fmt_out, false);
+            int r;
+
+            swr_close(s->avrctx);
+            if ((r = swr_init(s->avrctx)) < 0) {
+                close_lavrr(af);
+                return r;
```
Comment (wm4) [inline audio/filter/af_lavrresample.c]: No, mpv doesn't use ffmpeg error codes internally. Just don't return at all. close_lavrr sets avrctx to NULL, which will take appropriate actions when filtering is attempted.

Comment (richardpl) [inline audio/filter/af_lavrresample.c]: mpv segfaults if swr_init fails here, whatever I return...

**Pull request closed without merging**